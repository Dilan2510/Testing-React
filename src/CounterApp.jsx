import { useState } from "react";

export const CounterApp = ({ value }) => {
  const [count, setCount] = useState(value);

  function sumCount() {
    return setCount(count + 1);
  }
  function resCount() {
    return setCount(count - 1);
  }
  function resetCount() {
    return setCount(value);
  }

  return (
    <div>
      <h1>CounterApp</h1>
      <h2> {count} </h2>
      <button onClick={sumCount}>+1</button>
      <button onClick={resCount}>-1</button>
      <button aria-label="button-reset" onClick={resetCount}>
        Reset
      </button>
    </div>
  );
};
