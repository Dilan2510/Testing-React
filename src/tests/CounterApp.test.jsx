import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../CounterApp";

describe("se llama el component CounterApp", () => {
  const valueInitial = 1;
  test("testing for consumir el maatch en el snapshot> ", () => {
    const { container } = render(<CounterApp value={valueInitial} />);
    expect(container).toMatchSnapshot();
  });

  test("testing for view value initial from 100 ", () => {
    render(<CounterApp value={100} />);
    expect(screen.getByText(100)).toBeTruthy();
  });

  test("Debe imcrementar el valor del button mas 1 ", () => {
    render(<CounterApp value={valueInitial} />);
    fireEvent.click(screen.getByText("+1"));
    expect(screen.getByText("2")).toBeTruthy();
  });
  test("Debe disminuir el valor del button menos  1 ", () => {
    render(<CounterApp value={valueInitial} />);
    fireEvent.click(screen.getByText("-1"));
    expect(screen.getByText("0")).toBeTruthy();
  });
  test("Debe setear el valor del button  y quedar en 1 ", () => {
    render(<CounterApp value={valueInitial} />);
    fireEvent.click(screen.getByText(valueInitial));
    fireEvent.click(screen.getByRole("button", { name: "button-reset" }));
    expect(screen.getByText("1")).toBeTruthy();
  });
});
