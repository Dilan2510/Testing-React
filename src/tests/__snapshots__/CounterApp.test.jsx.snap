// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`se llama el component CounterApp testing for consumir el maatch en el snapshot>  1`] = `
<div>
  <div>
    <h1>
      CounterApp
    </h1>
    <h2>
       
      1
       
    </h2>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
    <button
      aria-label="button-reset"
    >
      Reset
    </button>
  </div>
</div>
`;
